server/resp/src/main/java/org/infinispan/server/resp/Resp3Handler.java

```diff
@@ -14,34 +14,44 @@
 import org.infinispan.commons.CacheException;
 import org.infinispan.commons.logging.LogFactory;
 import org.infinispan.server.core.logging.Log;
+import org.infinispan.server.core.transport.NativeTransport;
 import org.infinispan.util.concurrent.AggregateCompletionStage;
 import org.infinispan.util.concurrent.CompletionStages;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.util.CharsetUtil;
 
 public class Resp3Handler extends Resp3AuthHandler {
    private static final Log log = LogFactory.getLog(MethodHandles.lookup().lookupClass(), Log.class);
-   private static final ByteBuf OK = RespRequestHandler.stringToByteBuf("+OK\r\n", ByteBufAllocator.DEFAULT);
```

Comment:
We couldn't use DEFAULT byte buffer as we want to use a byte buf that is the same type as our underlying protocol. Since netty supports both native and JVM based sockets, any ByteBuf passed down that doesn't match the same type will be first copied into an unpooled copy in the appropriate version. Thus we need to make sure this is the same type as we would configure in our server.